shell/renderer/electron_renderer_client.cc

```diff
@@ -80,12 +80,22 @@ void ElectronRendererClient::DidCreateScriptContext(
   // TODO(zcbenz): Do not create Node environment if node integration is not
   // enabled.
 
-  // Only load node if we are a main frame or a devtools extension
-  // unless node support has been explicitly enabled for sub frames
+  // Only load Node.js if we are a main frame or a devtools extension
+  // unless Node.js support has been explicitly enabled for subframes.
   auto prefs = render_frame->GetBlinkPreferences();
   bool is_main_frame = render_frame->IsMainFrame();
   bool is_devtools = IsDevToolsExtension(render_frame);
-  bool allow_node_in_subframes = prefs.node_integration_in_sub_frames;
+
+  // This is necessary because if an iframe is created and a source is not
+  // set, the iframe loads about:blank and creates a script context for the
+  // same. We don't want to create a Node.js environment here because if the src
+  // is later set, the JS necessary to do that triggers illegal access errors
+  // when the initial about:blank Node.js environment is cleaned up. See:
+  // https://source.chromium.org/chromium/chromium/src/+/main:content/renderer/render_frame_impl.h;l=870-892;drc=4b6001440a18740b76a1c63fa2a002cc941db394
+  GURL url = render_frame->GetWebFrame()->GetDocument().Url();
+  bool allow_node_in_subframes =
```

Comment:
This check can be moved to https://github.com/electron/electron/blob/821c81f5ee09099a39157db06c0c5104f469cbbd/shell/renderer/electron_render_frame_observer.cc#L155 to avoid duplication for the sandboxed and non-sandboxed client code.

Reply:
Moved. We don't invoke script create context if `ShouldNotifyClient` does not satisfy, we can just avoid calling it for the `about:blank` there.